File: backend/sdk/java/src/main/java/com/waas/sdk/models/SendWebhookResponse.java
package com.waas.sdk.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.OffsetDateTime;
import java.util.UUID;

/**
 * Response from sending a webhook.
 */
public class SendWebhookResponse {
    @JsonProperty("delivery_id")
    private UUID deliveryId;
    
    @JsonProperty("endpoint_id")
    private UUID endpointId;
    
    private String status;
    
    @JsonProperty("scheduled_at")
    private OffsetDateTime scheduledAt;

    public UUID getDeliveryId() { return deliveryId; }
    public void setDeliveryId(UUID deliveryId) { this.deliveryId = deliveryId; }
    
    public UUID getEndpointId() { return endpointId; }
    public void setEndpointId(UUID endpointId) { this.endpointId = endpointId; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    
    public OffsetDateTime getScheduledAt() { return scheduledAt; }
    public void setScheduledAt(OffsetDateTime scheduledAt) { this.scheduledAt = scheduledAt; }
}

File: backend/sdk/php/src/Models/SendWebhookResponse.php
<?php

declare(strict_types=1);

namespace WAAS\Models;

use DateTimeImmutable;

class SendWebhookResponse
{
    public function __construct(
        public readonly string $deliveryId,
        public readonly string $endpointId,
        public readonly string $status,
        public readonly DateTimeImmutable $scheduledAt,
    ) {}

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            deliveryId: $data['delivery_id'],
            endpointId: $data['endpoint_id'],
            status: $data['status'],
            scheduledAt: new DateTimeImmutable($data['scheduled_at']),
        );
    }
}
